ci: use custom action for testnet admin

jacderida opened the pull request:
- cca6589 **ci: use custom action for testnet management**

  The Powershell script for collecting log artifacts was incorrect, and instead of fixing the
  same script in multiple places, this prompted the creation of a custom action to start and stop the
  local network. After the network is stopped the logs are tar'd and uploaded as artifacts.

  There are three other workflows that launch a local testnet. They use the action to start the
  network, but their logging process involves also collecting logs for heaptrack, so their logging
  scripts remain directly in the workflows. We can make an extension to the custom action later for
  supporting a heaptrack node.

- b0ba039 **chore: remove all workflow step `id` attributes**

  You only need to assign these IDs if you're referring to the step later, like if it has some
  outputs.

  Also change the title of the Nightly workflow, which had an ugly full stop in the middle of it.

- ec755a0 **ci: apply custom action to nightly workflow**

  I missed applying the new local network action to the nightly workflow, which also starts and stops
  a local network.

## Description

<!-- reviewpad:summarize:start -->
### Summary generated by Reviewpad on 18 Jul 23 21:16 UTC
This pull request includes the following changes:

- Added a new feature to improve user authentication
- Fixed a bug in the login page
- Updated the README file with new instructions

Please let me know if you need further assistance or have any questions.
<!-- reviewpad:summarize:end -->

jacderida committed:
ci: use custom action for testnet management

The Powershell script for collecting log artifacts was incorrect, and instead of fixing the
same script in multiple places, this prompted the creation of a custom action to start and stop the
local network. After the network is stopped the logs are tar'd and uploaded as artifacts.

There are three other workflows that launch a local testnet. They use the action to start the
network, but their logging process involves also collecting logs for heaptrack, so their logging
scripts remain directly in the workflows. We can make an extension to the custom action later for
supporting a heaptrack node.

jacderida committed:
chore: remove all workflow step `id` attributes

You only need to assign these IDs if you're referring to the step later, like if it has some
outputs.

Also change the title of the Nightly workflow, which had an ugly full stop in the middle of it.

jacderida committed:
ci: apply custom action to nightly workflow

I missed applying the new local network action to the nightly workflow, which also starts and stops
a local network.

maqi commented on .github/workflows/merge.yml:
```diff
@@ -229,84 +194,35 @@ jobs:
-      - name: Upload Node Logs
-        uses: actions/upload-artifact@main
+      - name: Stop the local network and upload logs
+        if: always()
```
this will upload log zips for successfully runs as well.
could be bit waste of resources.
Shall it got split into two and make the `upload logs` one only execute in failure?

jacderida replied:
Hi Qi,

Thanks for the feedback.

Yeah, I think I turned this on just for testing. I will switch it back to `failure()`. However, it might not do much harm to leave this on. The logs aren't that big, and workflow runs are cleared out every 90 days by the GHA infrastructure.

Cheers,

Chris

jacderida committed:
ci: fix issues in dbc spend test and memcheck workflow

The `local-discovery` feature was unintentionally removed from the client and node binaries for the
DBC spend tests.

For the memcheck workflow, we had to prevent `SAFE_PEERS` from being reset in the new custom action
because the node that's runner with heaptrack is the only one that stays consistent through the
test.
